src/ts/message.ts

```diff
@@ -184,12 +184,12 @@ export function printMessage(fileName: string, exportMap: ExportMap, messageDesc
         printer.printIndentedLn(`get${withUppercase}List(): Array<Uint8Array | string>;`);
         printer.printIndentedLn(`get${withUppercase}List_asU8(): Array<Uint8Array>;`);
         printer.printIndentedLn(`get${withUppercase}List_asB64(): Array<string>;`);
-        printer.printIndentedLn(`set${withUppercase}List(value: Array<Uint8Array | string>): void;`);
+        printer.printIndentedLn(`set${withUppercase}List(value: Array<Uint8Array | string>): ${messageName};`);
```

Comment:
From the Protoc docs: 
> addFoo(): Appends a value of foo to the end of the list of foos that was in the message. Returns the outer message for chaining only if the added value was a primitive. For added messages, returns the message that was added (b/138079947).

https://developers.google.com/protocol-buffers/docs/reference/javascript-generated

My understanding is this means the return type of `setInternalChildMessage` for example should be the child message type, and not the parent message type. 

```typescript
        // Wrong 
        setInternalChildMessage(value?: OneOfMessage.InternalChildMessage): OneOfMessage;
        
        // Correct
        setInternalChildMessage(value?: OneOfMessage.InternalChildMessage): OneOfMessage.InternalChildMessage;
```

Can you add some test cases to verify this?